avocado/core/nrunner.py

```diff
@@ -240,23 +298,108 @@ def task_from_recipe(task_path):
     runnable = Runnable(runnable_recipe.get('kind'),
                         runnable_recipe.get('uri'),
                         *runnable_recipe.get('args', ()))
-    return Task(identifier, runnable)
+    status_uris = recipe.get('status_uris')
+    return Task(identifier, runnable, status_uris)
+
+
+class StatusServer:
+
+    def __init__(self, uri, tasks_pending=None):
+        self.uri = uri
+        self.server_task = None
+        self.status = {}
+        if tasks_pending is None:
+            tasks_pending = []
+        self.tasks_pending = tasks_pending
+        self.wait_on_tasks_pending = len(self.tasks_pending) > 0
+
+    @asyncio.coroutine
+    def cb(self, reader, _):
+        while True:
+            if self.wait_on_tasks_pending:
```

Comment:
Is your preference use nested `ifs` instead of single `if` with `and` conditional?

`if self.wait_on_tasks_pending and not self.tasks_pending`